rapthor/lib/field.py

```diff
@@ -976,16 +981,69 @@ def makeWCS(self):
         w.wcs.set_pv([(2, 1, 45.0)])
         self.wcs = w
 
-    def check_selfcal_convergence(self):
+    def check_selfcal_progress(self, convergence_ratio=None, divergence_ratio=None):
```

Comment:
Any particular reason why you use class attributes `self.convergence_ratio` and `self.divergence_ratio` to set `convergence_ratio` and `divergence_ratio` when they're not set? Another option would be to set those values as defaults here, instead of using `None`.

Reply:
To be honest, I can't remember why I did this...  At any rate, I will remove the two ratios from the args and just get the values from the class attributes, as I can't think of a need to have them settable any other way.